lib/compress.js

```diff
@@ -2989,33 +2989,30 @@ merge(Compressor.prototype, {
         return self;
     });
 
-    function has_side_effects_or_prop_access(node, compressor) {
-        var save_pure_getters = compressor.option("pure_getters");
-        compressor.options.pure_getters = false;
-        var ret = node.has_side_effects(compressor);
-        compressor.options.pure_getters = save_pure_getters;
-        return ret;
-    }
-
     AST_Binary.DEFMETHOD("lift_sequences", function(compressor){
         if (compressor.option("sequences")) {
             if (this.left instanceof AST_Seq) {
                 var seq = this.left;
                 var x = seq.to_array();
                 this.left = x.pop();
                 x.push(this);
-                seq = AST_Seq.from_array(x).transform(compressor);
-                return seq;
-            }
-            if (this.right instanceof AST_Seq
-                && this instanceof AST_Assign
-                && !has_side_effects_or_prop_access(this.left, compressor)) {
-                var seq = this.right;
-                var x = seq.to_array();
-                this.right = x.pop();
-                x.push(this);
-                seq = AST_Seq.from_array(x).transform(compressor);
-                return seq;
+                return AST_Seq.from_array(x).optimize(compressor);
+            }
+            if (this.right instanceof AST_Seq && !this.left.has_side_effects(compressor)) {
+                var assign = this.operator == "=" && this.left instanceof AST_SymbolRef;
+                var root = this.right;
+                var cursor, seq = root;
+                while (assign || !seq.car.has_side_effects(compressor)) {
```

Comment:
`assign` is the original (and now fixed) functionality. I've also introduced a new rule whereby if LHS of `AST_Binary` and that of `AST_Seq` are both side-effect-free, we can percolate the former onto the RHS of the latter.